doc/apiv3-documentation.apib

```diff
@@ -3153,6 +3162,94 @@ Permanently deletes the specified user account.
+| user      | The user that this preference belongs to                 | User           | not null              | READ                 |
+
+## Local Properties
+| Property               | Description                                                | Type       | Constraints | Supported operations |
+|:----------------------:| -----------------------------------------------------------| ---------- | ----------- | -------------------- |
+| hideMail               | Hide mail address from other users                         | Boolean    |             | READ / WRITE         |
+| timeZone               | Current selected time zone                                 | String     |             | READ / WRITE         |
+| theme                  | Current selected theme                                     | String     |             | READ / WRITE         |
+| commentSortDescending  | Sort comments in descending order                          | Boolean    |             | READ / WRITE         |
+| warnOnLeavingUnsaved   | Issue warning when leaving a page with unsaved text        | Boolean    |             | READ / WRITE         |
+| accessibilityMode      | Enable accessibility mode                                  | Boolean    |             | READ / WRITE         |
+
+## UserPreferences [/api/v3/my_preferences]
+
++ Model
+    + Body
+
+            {
+                "_type" : "UserPreferences",
+                "_links" : {
+                    "self" : {
+                        "href" : "/api/v3/my_preferences",
+                    },
+                    "user": {
+                        "href": "/api/v3/users/1",
+                        "title": "John Sheppard"
+                    }
+                },
+                "hideMail" : false,
+                "timeZone" : "Europe/Berlin",
+                "theme" : "OpenProject",
+                "commentSortDescending" : true,
+                "warnOnLeavingUnsaved" : true,
+                "accessibilityMode" : false
+            }
+
+
+## Show my preferences [GET]
+
++ Response 200 (application/hal+json)
+
+    [UserPreferences][]
+
++ Response 401 (application/hal+json)
+
+    Returned if no user is currently authenticated
+
+    + Body
+
+            {
+                "_type": "Error",
+                "errorIdentifier": "urn:openproject-org:api:v3:errors:Unauthenticated",
+                "message": "You need to be authenticated to access this resource."
+            }
+
+## Update UserPreferences [PATCH]
+
+When calling this endpoint the client provides a single object, containing the properties that it wants to change, in the body.
+
++ Request (application/json)
+
+        {
+            "accessibilityMode": true,
+            "timeZone": "Europe/Paris"
+        }
+
++ Response 200 (application/hal+json)
+
+    [UserPreferences][]
+
++ Response 401 (application/hal+json)
```

Comment:
I guess missing response codes are:

* 400 (for malformed JSON/syntax errors etc)
* 422 (for semantically wrong data, e.g. `timeZone: middleearth/mordor`)

Not sure if you will be able to detect the latter cases, though